spec/ndx-patterned-ogen.extensions.yaml

```diff
@@ -1,8 +1,165 @@
+        required: false
+      - name: number_of_revolutions
+        dtype: numeric
+        doc: Number of turns within a spiral.
+  - neurodata_type_def: TemporalFocusing
+    neurodata_type_inc: LabMetaData
+    doc: Container to store the parameters defining a temporal focusing beam-shaping.
+    attributes:
+      - name: description
+        dtype: text
+        doc: Description of the pattern.
+        required: false
+      - name: lateral_point_spread_function
+        dtype: text
+        doc:
+          Estimated lateral spatial profile or point spread function, expressed as
+          mean [um] ± s.d [um].
+      - name: axial_point_spread_function
+        dtype: text
+        doc:
+          Estimated axial spatial profile or point spread function, expressed as mean
+          [um]± s.d [um].
+  - neurodata_type_def: PatternedOptogeneticStimulusSite
+    neurodata_type_inc: OptogeneticStimulusSite
+    doc: Patterned optogenetic stimulus site.
+    attributes:
+      - name: effector
+        dtype: text
+        doc: Light-activated effector protein expressed by the targeted cell (e.g., ChR2).
+    links:
+      - name: spatial_light_modulator
+        target_type: SpatialLightModulator
+        doc: SpatialLightModulator used to generate holographic pattern.
+      - name: light_source
+        target_type: LightSource
+        doc: LightSource used to apply photostimulation.
+  - neurodata_type_def: SpatialLightModulator
+    neurodata_type_inc: Device
+    doc: Spatial light modulator used in the experiment.
+    attributes:
+      - name: model
+        dtype: text
+        doc: Model of SpatialLightModulator.
+        required: false
+      - name: size
+        dtype: numeric
+        doc:
+          Resolution of SpatialLightModulator (in pixels), formatted as [width, height]
+          or [width, height, depth].
+        required: false
+  - neurodata_type_def: LightSource
+    neurodata_type_inc: Device
+    doc: Light source used in the experiment.
+    attributes:
+      - name: model
+        dtype: text
+        doc: Model of LightSource.
+        required: false
+      - name: stimulation_wavelength
+        dtype: numeric
+        doc: Excitation wavelength of stimulation light (nanometers).
+      - name: peak_power
+        dtype: numeric
+        doc: Incident power of stimulation device (in Watts).
+        required: false
+      - name: peak_pulse_energy
+        dtype: numeric
+        doc: If device is pulsed laser, pulse energy (in Joules).
+        required: false
+      - name: intensity
+        dtype: numeric
+        doc: Intensity of the excitation in W/m^2, if known.
+        required: false
+      - name: exposure_time
+        dtype: numeric
+        doc: exposure time of the sample
+        required: false
+      - name: pulse_rate
+        dtype: numeric
+        doc: If device is pulsed laser, pulse rate (in Hz) used for stimulation.
+        required: false
```

Comment:
Would `pulse_rate` apply to any other general light source or is it, as the doc describes, laser-specific? If laser-specific, I would just add it to the laser subtype

Reply:
Similarly applies to above laser-referenced values on the parent

Reply:
Ah no sorry, it should be pulsed light source in general. LED could be used in a "pulsed" configuration too. 
Good catch!